wdae/wdae/datasets_api/permissions.py

```diff
@@ -43,29 +48,174 @@ def has_permission(self, request: HttpRequest, view: Any) -> bool:
         return self.has_object_permission(request, view, dataset_id)
 
     def has_object_permission(
-            self, request: HttpRequest, view: Any, obj: str,
+        self, request: HttpRequest, _view: Any, obj: str,
     ) -> bool:
         wgpf_instance = get_wgpf_instance()
-        if user_has_permission(
+        return user_has_permission(
             wgpf_instance.instance_id, cast(User, request.user), obj,
-        ):
-            return True
+        )
 
-        return False
+    @staticmethod
+    def get_allowed_datasets_query() -> str:
```

Comment:
`get_allowed_` is too broad... consider using `construct` or `build` or `prepare` instead of `get`